django/db/models/sql/query.py

```diff
@@ -534,8 +534,14 @@ def exists(self):
                 # SELECT clause which is about to be cleared.
                 q.set_group_by(allow_aliases=False)
             q.clear_select_clause()
+        if q.combined_queries and q.combinator == 'union':
+            limit_combined = connections[DEFAULT_DB_ALIAS].features.supports_limiting_in_compound
```

Comment:
I think using `connections[DEFAULT_DB_ALIAS]` is acceptable here as we even do it at a higher level (e.g. feature detection in `django.db.models`) but if we truly wanted to perform the check against a database the following minimal changes would allow you to use the proper database alias

```diff
diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
index c596ccfe5d..e356a5f981 100644
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -1116,10 +1116,10 @@ def copy(self):
     def external_aliases(self):
         return self.query.external_aliases

-    def as_sql(self, compiler, connection, template=None, **extra_context):
+    def as_sql(self, compiler, connection, query=None, template=None, **extra_context):
         connection.ops.check_expression_support(self)
         template_params = {**self.extra, **extra_context}
-        subquery_sql, sql_params = self.query.as_sql(compiler, connection)
+        subquery_sql, sql_params = (query or self.query).as_sql(compiler, connection)
         template_params['subquery'] = subquery_sql[1:-1]

         template = template or template_params.get('template', self.template)
@@ -1142,7 +1142,6 @@ class Exists(Subquery):
     def __init__(self, queryset, negated=False, **kwargs):
         self.negated = negated
         super().__init__(queryset, **kwargs)
-        self.query = self.query.exists()

     def __invert__(self):
         clone = self.copy()
@@ -1150,7 +1149,9 @@ def __invert__(self):
         return clone

     def as_sql(self, compiler, connection, template=None, **extra_context):
-        sql, params = super().as_sql(compiler, connection, template, **extra_context)
+        sql, params = super().as_sql(
+            compiler, connection, template, query=self.query.exists(connection.alias), **extra_context
+        )
         if self.negated:
             sql = 'NOT {}'.format(sql)
         return sql, params
diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
index a7dadf5a40..0d29aa28f3 100644
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -525,7 +525,7 @@ def get_count(self, using):
     def has_filters(self):
         return self.where

-    def exists(self):
+    def exists(self, using):
         q = self.clone()
         if not q.distinct:
             if q.group_by is True:
@@ -541,7 +541,7 @@ def exists(self):
         return q

     def has_results(self, using):
-        q = self.exists()
+        q = self.exists(using)
         compiler = q.get_compiler(using=using)
         return compiler.has_results()
```